pkg/controller/persistentvolume/controller_test.go

```diff
@@ -148,13 +149,17 @@ func TestControllerSync(t *testing.T) {
 			time.Sleep(10 * time.Millisecond)
 		}
 
+		count := reactor.getChangeCount()
+
 		// Call the tested function
 		err := test.test(ctrl, reactor, test)
 		if err != nil {
 			t.Errorf("Test %q initial test call failed: %v", test.name, err)
 		}
 
-		reactor.waitTest()
+		for reactor.getChangeCount() < count+expectedChanges[ix] {
```

Comment:
@brendandburns Drive-by comment.  What prevents this from entering an endless loop?